lib/model/devicedownloadstate.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) 2015 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package model
+
+import (
+	"github.com/syncthing/syncthing/lib/protocol"
+	"github.com/syncthing/syncthing/lib/sync"
+)
+
+type deviceFolderFileDownloadState struct {
+	blockIndexes []int32
```

Comment:
Do we really want to go through the effort of type asserting?
This is an internal field so nobody should care about the type anyway.